### ISSUE TITLE
Introduce new macros to simplify declaration of parallel loops

### ISSUE THREAD

**Issue by mikerainey** [deepsea-inria/encore#18]
Before:

  static constexpr int nb_loops = 1;
  encore_private_activation_record_begin(encore::edsl, maxIndex, nb_loops)
    int s; int e;
  encore_private_activation_record_end(encore::edsl, maxIndex, sar, par, dc, get_dc)

After:

  encore_private_activation_record_begin(encore::edsl, maxIndex)
  encore_parallel_for_declare()
    int s; int e;
  encore_private_activation_record_end(encore::edsl, maxIndex, sar, par, dc, get_dc)

In other words, declare each loop inline by a macro call, such as encore_parallel_for_declare(), encore_parallel_combine_declare(), etc.
